### PR TITLE
Make some tweaks to RegExp debug switches.

### PR DESCRIPTION
**Pull request by dilijev**
This change only affects debug output displayed under `-RegexDebug`.

Prior to this change, `-RegexDebug` has 3 "chapters" of output, all enabled with no option to disable: AST, Annotated AST, and (Bytecode) Program.

This change adds flags `-RegexDebugAST[-]` and `-RegexDebugAnnotatedAST[-]`  to disable the AST output chapters and bring focus to the resulting bytecode program, which is the most relevant for seeing how effective the regex bytecode gen was for a given case. The AST output can be noisy and distracts from this goal.

This change leaves the AST output chapters on-by-default to match existing behavior.

The AST may be interesting information when working on the Regex engine even if not doing work on optimizations. In that case the on-by-default behavior for `-RegexDebug` should probably be: `-RegexDebugAST -RegexDebugAnnotatedAST-`

### CODE REVIEW

File: lib/Common/ConfigFlagsList.h [hunk 1]
```diff
@@ -558,6 +558,8 @@ PHASE(All)
 #define DEFAULT_CONFIG_RegexTracing         (false)
 #define DEFAULT_CONFIG_RegexProfile         (false)
 #define DEFAULT_CONFIG_RegexDebug           (false)
+#define DEFAULT_CONFIG_RegexDebugAST        (true)
+#define DEFAULT_CONFIG_RegexDebugAnnotatedAST (true)
```
Comment (kfarnung) [inline lib/Common/ConfigFlagsList.h]: Should these be true by default?

Comment (dilijev) [reply lib/Common/ConfigFlagsList.h]: On-by-default for these is the current behavior.

Right now -RegexDebug has 3 "chapters" of output. AST, Annotated AST, and (Bytecode) Program.

Personally I'm usually more interested in reviewing the bytecode program than I am in debugging the AST so I've always wanted to have an option to turn these output segments off (or, at least the Annotated AST, but it was easy to also disable the other chapter while I was in the area). This change adds such flags.

I'd be tempted to turn off the annotated AST which is a ton of output for not a lot of value (unless you're actively working on AST-based optimizations in the regex engine), and would probably vote for leaving basic AST on by default.

Comment (kfarnung) [reply lib/Common/ConfigFlagsList.h]: I see, so these are already behind the RegexDebug flag.

Comment (dilijev) [reply lib/Common/ConfigFlagsList.h]: Yep!

**Comment by dilijev** [comment]
@dotnet-bot test this please

**Comment by dilijev** [comment]
@dotnet-bot test this please

**Comment by dilijev** [comment]
@dotnet-bot 
test Windows 8.1 arm_test
test Windows 8.1 ci_legacy8_x64_debug

**Pull request merged**